src/main/java/seedu/address/model/person/AddressContainsKeywordsPredicate.java

```diff
@@ -0,0 +1,51 @@
+package seedu.address.model.person;
+
+import java.util.List;
+import java.util.function.Predicate;
+
+import seedu.address.commons.util.ToStringBuilder;
+
+/**
+ * Tests that a {@code Person}'s {@code Phone} matches any of the keywords given.
+ */
+public class AddressContainsKeywordsPredicate implements Predicate<Person> {
+    private final List<String> keywords;
+
+    public AddressContainsKeywordsPredicate(List<String> keywords) {
+        this.keywords = keywords;
+    }
+
+    @Override
+    public boolean test(Person person) {
+        // Checks if the string i.e (address) contains a keyword, allowing partial matching of address via find command
+        return keywords.stream()
+                .anyMatch(keyword -> person.getAddress().value.toLowerCase().contains(keyword.toLowerCase()));
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        System.out.println("This is in equals of AddressContainsPredicate class!");
+        System.out.println(other);
```

Comment:
I think this is for debugging? Can remove, if it is important, I suggest to using `Logger` for consistency